python/pylibcudf/pylibcudf/io/types.pyx

```diff
@@ -74,18 +80,24 @@ cdef class ColumnInMetadata:
     Metadata for a column
     """
 
+    def __init__(self):
+        raise ValueError(
+            "ColumnInMetadata should not be constructed directly. "
+            "Use one of the factories."
+        )
+
     @staticmethod
-    cdef ColumnInMetadata from_metadata(column_in_metadata metadata):
+    cdef ColumnInMetadata from_libcudf_ptr(column_in_metadata* metadata):
         """
-        Construct a ColumnInMetadata.
+        Create a Python ColumnInMetadata from a libcudf column_in_metadata pointer.
 
         Parameters
         ----------
-        metadata : column_in_metadata
+        metadata : column_in_metadata*
         """
-        cdef ColumnInMetadata col_metadata = ColumnInMetadata.__new__(ColumnInMetadata)
-        col_metadata.c_obj = metadata
-        return col_metadata
+        cdef ColumnInMetadata out = ColumnInMetadata.__new__(ColumnInMetadata)
+        out.c_obj = metadata
+        return out
```

Comment:
To match up with the usual conventions, let's call this `from_libcudf`. And, to make it more difficult to misuse, let us also accept the owning tableinputmetadata:

```
cdef ColumnInMetadata from_libcudf(
    column_in_metadata* metadata, TableInputMetadata owner not None
):
    """
    A Python representation of `column_in_metadata`.

    Parameters
    ----------
    metadata : column_in_metadata*
        Raw pointer to C++ metadata.
    owner : TableInputMetadata
        Owning table input metadata that manages lifetime of the raw pointer.
    """
    cdef ColumnInMetadata self = ColumnInMetadata.__new__(ColumnInMetadata)
    self.c_obj = metadata
    self.owner = owner
    return self
```